src/foundry/common/grid/hexagonal.d.mts

```diff
@@ -0,0 +1,392 @@
+
+  /**
+   * Compute the grid configuration from a provided type
+   * @param type - The grid type
+   * @param size - The grid size in pixels
+   * @deprecated Since v12 until v14. No Replacement
+   */
+  static getConfig(type: number, size: number): HexagonalGrid.Configuration;
+
+  /**
+   * Convert an offset coordinate (row, col) into a cube coordinate (q, r, s).
+   * See https://www.redblobgames.com/grids/hexagons/ for reference
+   * Source code available https://www.redblobgames.com/grids/hexagons/codegen/output/lib-functions.js
+   * @param offset - The offset coordinate
+   * @param config - The hex grid configuration
+   * @deprecated Since v12 until v14. Use {@link HexagonalGrid#offsetToCube} instead.
+   */
+  static offsetToCube(
+    offset: HexOffsetCoordinate,
+    config?: InexactPartial<{
+      columns: boolean;
+      even: boolean;
+    }>,
+  ): HexagonalGrid.Cube;
+
+  /**
+   * Convert a cube coordinate (q, r, s) into an offset coordinate (row, col).
+   * See https://www.redblobgames.com/grids/hexagons/ for reference
+   * Source code available https://www.redblobgames.com/grids/hexagons/codegen/output/lib-functions.js
+   * @param cube   - The cube coordinate
+   * @param config - The hex grid configuration
+   * @returns The offset coordinate
+   * @deprecated Since v12 until v14. Use {@link HexagonalGrid#cubeToOffset} instead.
+   */
+  static cubeToOffset(cube: HexCubeCoordinate, config: HexGridConfiguration): HexagonalGrid.Offset;
+
+  /**
+   * Given a cursor position (x, y), obtain the cube coordinate hex (q, r, s) of the hex which contains it
+   * http://justinpombrio.net/programming/2020/04/28/pixel-to-hex.html
+   * @param point  - The pixel point
+   * @param config - The hex grid configuration
+   * @returns The cube coordinate
+   * @deprecated Since v12 until v14. Use {@link HexagonalGrid#pointToCube} instead.
+   */
+  static pixelToCube(point: Point, config: HexGridConfiguration): HexagonalGrid.Coordinates;
+
+  /**
+   * Compute the top-left pixel coordinate of a hexagon from its offset coordinate.
+   * @param offset - The offset coordinate
+   * @param config - The hex grid configuration
+   * @returns The coordinate in pixels
+   * @deprecated Since v12 until v14. Use {@link HexagonalGrid#getTopLeftPoint} instead.
+   */
+  static offsetToPixels(offset: HexOffsetCoordinate, config: HexGridConfiguration): Point;
+
+  /**
+   * Compute the offset coordinate of a hexagon from a pixel coordinate contained within that hex.
+   * @param point  - The pixel coordinate
+   * @param config - The hex grid configuration
+   * @param method - Which Math rounding method to use
+   *                 (default: "floor")
+   * @returns The offset coordinate
+   * @deprecated Since v12 until v14. No Replacement
+   */
+  static pixelsToOffset(point: Point, config: HexGridConfiguration, method?: keyof Math): HexagonalGrid.Coordinates;
```

Comment:
Hmm this is deprecated but note that this would allow something like `"hasOwnProperty"` which is clearly nonsensical. Just to note for the future.